service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapperTest.java

```diff
@@ -293,6 +297,61 @@ void When_DiagnosticReport_With_NoReferencedSpecimenAndFilingCommentWithNoCommen
         assertThat(actualXml).isEqualToIgnoringWhitespace(expectedXml);
     }
 
+    /**
+     * A Diagnosis Report may have an Observation (Test Result) and Specimen. If the result and specimen are not
+     * linked then we need to create a dummy specimen linked to the result.
+     */
+    @Test
+    void When_DiagnosticReport_Has_SpecimenAndUnlinkedTestResult_Expect_ADummySpecimenLinkedToTestResult() {
+        final String diagnosticReportFileName = "diagnostic-report-with-one-specimen-and-one-unrelated-observation.json";
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(diagnosticReportFileName);
+        final Bundle bundle = getBundleResourceFromJson(INPUT_JSON_BUNDLE);
+        final InputBundle inputBundle = new InputBundle(bundle);
+        final String expectObservation = "Observation/TestResult-WithoutSpecimenReference";
+        final String dummyNamePrefix = DiagnosticReportMapper.DUMMY_SPECIMEN_ID_PREFIX;
+
+        when(messageContext.getInputBundleHolder()).thenReturn(inputBundle);
+
+        final String actualXml = mapper.mapDiagnosticReportToCompoundStatement(diagnosticReport);
+
+        // This checks that the unlinked test result is given a dummy specimen.
+        assertThat(actualXml).matches("(?s).*<!-- " + MOCKED_SPECIMEN_PREFIX + ": " + dummyNamePrefix
+                + REGEXP_UUID
+                + " " + MOCKED_SPECIMEN_LINKED_OBSERVATION + ": "
+                + expectObservation
+                + "-->.*"
+        );
```

Comment:
This could be simplified. No need to use a regular expression. Also, if we keep this test then the associated `diagnostic-report-with-one-specimen-and-one-unrelated-observation.xml` test can be removed.

Reply:
Correct. Dealt with.